src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -0,0 +1,708 @@
+        Delegate d2 = (ref object x2) => { _ = x2.Length; };
+        Delegate d3 = delegate (object x3) { _ = x3.Length; };
+    }
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics(
+                // (6,23): error CS8915: The delegate type could not be inferred.
+                //         Delegate d0 = x0 => { _ = x0.Length; };
+                Diagnostic(ErrorCode.ERR_CannotInferDelegateType, "x0 => { _ = x0.Length; }").WithLocation(6, 23),
+                // (6,38): error CS1061: 'object' does not contain a definition for 'Length' and no accessible extension method 'Length' accepting a first argument of type 'object' could be found (are you missing a using directive or an assembly reference?)
+                //         Delegate d0 = x0 => { _ = x0.Length; };
+                Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, "Length").WithArguments("object", "Length").WithLocation(6, 38),
+                // (7,47): error CS1061: 'object' does not contain a definition for 'Length' and no accessible extension method 'Length' accepting a first argument of type 'object' could be found (are you missing a using directive or an assembly reference?)
+                //         Delegate d1 = (object x1) => { _ = x1.Length; };
+                Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, "Length").WithArguments("object", "Length").WithLocation(7, 47),
+                // (8,23): error CS8915: The delegate type could not be inferred.
+                //         Delegate d2 = (ref object x2) => { _ = x2.Length; };
+                Diagnostic(ErrorCode.ERR_CannotInferDelegateType, "(ref object x2) => { _ = x2.Length; }").WithLocation(8, 23),
+                // (8,51): error CS1061: 'object' does not contain a definition for 'Length' and no accessible extension method 'Length' accepting a first argument of type 'object' could be found (are you missing a using directive or an assembly reference?)
+                //         Delegate d2 = (ref object x2) => { _ = x2.Length; };
+                Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, "Length").WithArguments("object", "Length").WithLocation(8, 51),
+                // (9,53): error CS1061: 'object' does not contain a definition for 'Length' and no accessible extension method 'Length' accepting a first argument of type 'object' could be found (are you missing a using directive or an assembly reference?)
+                //         Delegate d3 = delegate (object x3) { _ = x3.Length; };
+                Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, "Length").WithArguments("object", "Length").WithLocation(9, 53));
+        }
+
+        [Fact]
+        public void InstanceMethods()
+        {
+            var source =
+@"using System;
+class Program
+{
+    object F1() => null;
+    void F2(object x, int y) { }
+    void F()
+    {
+        Delegate d1 = F1;
+        Delegate d2 = this.F2;
```

Comment:
It looks like `base` has special rules for binding method groups. Consider adding a test:
See BindInstanceMemberAccess (line 6290)
`var searchExtensionMethodsIfNecessary = !leftIsBaseReference;`

Reply:
Consider also testing with type as LHS: `Delegate d3 = Program3.StaticMethod` and `Delegate d4 = Program4<int>.StaticMethod`

---
In reply to: [612866844](https://github.com/dotnet/roslyn/pull/52448#discussion_r612866844) [](ancestors = 612866844)

Reply:
Added `ExtensionMethods_05()` for `base` with an extension method; added tests for static methods with explicit type to `GetMethodGroupData()`; added `DynamicReceiver()` for `dynamic` test.